File: Relay_file/relay_test_1.c
#include <wiringPi.h>
#include <stdio.h>
#include <stdlib.h>

#define RELAY_PIN 5  // wiringPi 핀 번호: GPIO24는 wiringPi에서 5번

int main(void) {
    int value;
    int relay_on = 0;

    // wiringPi 초기화
    if (wiringPiSetup() == -1) {
        printf("wiringPi initialization fail!\n");
        return 1;
    }

    pinMode(RELAY_PIN, OUTPUT);
    digitalWrite(RELAY_PIN, HIGH);  // 릴레이 OFF (HIGH가 OFF인 모듈 기준)

    while (1) {
        printf("Input sensor value: ");
        scanf("%d", &value);

        if (!relay_on && value >= 50) {
            printf("Relay ON\n");
            digitalWrite(RELAY_PIN, LOW);  // 릴레이 ON
            relay_on = 1;
        }
        else if (relay_on && value <= 30) {
            printf("Relay OFF\n");
            digitalWrite(RELAY_PIN, HIGH);  // 릴레이 OFF
            relay_on = 0;
        }
        else {
            printf("Relay on ? : %s\n", relay_on ? "ON" : "OFF");
        }

        delay(4000);  // 1초 대기
    }

    return 0;
}

File: Relay_file/relay_test_1.py
import RPi.GPIO as GPIO
import time

RELAY_PIN = 24  # ext-GPIO0 => GPIO24
GPIO.setmode(GPIO.BCM)
GPIO.setup(RELAY_PIN, GPIO.OUT)

relay_on = False  # 현재 릴레이 상태 추적

try:
    while True:
        # 여기에 센서값을 읽는 부분이 들어올 예정이지만,
        # 지금은 수동으로 테스트용 값 입력
        value = int(input("센서 값을 입력하세요: "))  # 예: 온도나 습도 값
        
        if not relay_on and value >= 50:
            print("릴레이 ON")
            GPIO.output(RELAY_PIN, GPIO.LOW)  # 릴레이 ON
            relay_on = True

        elif relay_on and value <= 30:
            print("릴레이 OFF")
            GPIO.output(RELAY_PIN, GPIO.HIGH)  # 릴레이 OFF
            relay_on = False

        else:
            print("릴레이 상태 유지:", "ON" if relay_on else "OFF")

        time.sleep(1)

except KeyboardInterrupt:
    print("프로그램 종료")

finally:
    GPIO.cleanup()
